### PR TITLE
Implemented like dislike endpoints with prometheus metrics

### LINKED ISSUE #216: Implement REST API endpoint to like a rule analysis result
According to https://github.com/RedHatInsights/insights-results-aggregator/issues/43 we'll need an endpoint to be able to like a rule by any user for a cluster.

### LINKED ISSUE #217: Implement REST API endpoint to dislike a rule analysis result
According to https://github.com/RedHatInsights/insights-results-aggregator/issues/43 we'll need an endpoint to be able to dislike a rule by any user for a cluster.

### LINKED ISSUE #245: Create prometheus metrics for user's feedback
Create prometheus metrics for user's feedback(likes, dislikes, messages)

### LINKED ISSUE #253: Implement storage methods for user's feedback on rules(like, dislike)
As a programmer I expect storage to implement methods to do next things:

1 store like for given cluster, user and rule
2 store dislike for given cluster, user and rule
3 store no vote for given cluster, user and rule(to reset user's vote)
4 store text feedback for given cluster, user and rule
5 get latest feedback for given cluster, user and rule

### PR DESCRIPTION
**Pull request by Sergey1011010**
# Description

implemented like dislike endpoints with prometheus metrics

Fixes #216 
Fixes #217
Fixes #245 
Fixes #253

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update
- Refactor (refactoring code, removing useless files)

## Testing steps

`make before_commit`

### CODE REVIEW

**Comment by natiiix** [comment]
Should we join these changes together somehow or should I make mine the third migration?

**Comment by Sergey1011010** [comment]
@natiiix one who completes his pull request first will win second migration :)

File: migration/migration.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ type Migration struct {
 // create the most recent version of the database from scratch.
 var migrations = []Migration{
 	mig1,
-	mig2,
+	mig3,
```
Comment (natiiix) [inline migration/migration.go]: Please make sure to add the `mig2`, otherwise you'd just replace my `mig2` with `mig3`.

File: server/server.go [hunk 2]
```diff
@@ -166,17 +216,24 @@ func (server *HTTPServer) Initialize(address string) http.Handler {
...
+
 	// common REST API endpoints
-	router.HandleFunc(server.Config.APIPrefix, server.mainEndpoint).Methods("GET")
-	router.HandleFunc(server.Config.APIPrefix+"organizations", server.listOfOrganizations).Methods("GET")
-	router.HandleFunc(server.Config.APIPrefix+"organizations/{organization}/clusters", server.listOfClustersForOrganization).Methods("GET")
-	router.HandleFunc(server.Config.APIPrefix+"report/{organization}/{cluster}", server.readReportForCluster).Methods("GET")
+	router.HandleFunc(apiPrefix, server.mainEndpoint).Methods("GET")
+	router.HandleFunc(apiPrefix+"organizations", server.listOfOrganizations).Methods("GET")
+	router.HandleFunc(apiPrefix+"report/{organization}/{cluster}", server.readReportForCluster).Methods("GET")
+	router.HandleFunc(apiPrefix+"rule/{cluster}/{rule_id}/like", server.likeRule).Methods("PUT")
```
Comment (quarckster) [inline server/server.go]: Please follow API designing guidelines https://platform-docs.cloud.paas.psi.redhat.com/rest/urls.html
I should look like this:
`clusters/{cluster_id}/rules/{rule_id}/like`

Comment (Sergey1011010) [reply server/server.go]: fixed

**Pull request merged**